interaction-manager/src/lib_components/spatial-navigation.js

```diff
@@ -279,38 +272,55 @@ class SpatialNavigation {
         });
     }
 
+    /**
+     * Resets to the original keys state
+     */
+    resetKeys() {
+        this.removeKeyActions();
+        this.activeKeys = JSON.parse(JSON.stringify(defaultKeysState));
+        this.registerKeyActions();
+    }
+
     /**
      * Adds or override default direction keys with the specified ones
      * @param {Object} customDirections - { up: 'W', left: 'A', right: 'D', down: 'S' }
      * @param {Object} options - Optional settings.
-     * @param {Boolean} options.changeDefault - If true, overrides the default navigation keys. Defaults to false.
+     * @param {Boolean} options.clearCurrentActiveKeys - If true, overrides all keys. Defaults to false.
      * @returns {void}
      */
-    changeKeys(customDirections, options = { changeDefault: false }) {
-        if (Object.entries(customDirections).length === 0) return;
+    changeKeys(customDirections, options = { clearCurrentActiveKeys: false }) {
+        const customKeysDirections = Object.keys(customDirections);
+        if (customKeysDirections.length === 0) return;
 
-        this.changeDefault = options.changeDefault;
-
-        const incorrectDirections = Object.keys(customDirections).filter(direction => !directions.includes(direction));
+        const incorrectDirections = customKeysDirections.filter(direction => !directions.includes(direction));
         if (incorrectDirections.length > 0) return console.error(`The following directions: [${incorrectDirections.join(', ')}] you have entered are incorrect! `);
 
-        this.customKeys = {
-            ...this.customKeys,
-            ...customDirections,
-        };
+        this.clearCurrentActiveKeys = options.clearCurrentActiveKeys;
+
+        if (this.registeredKeys.size !== 0) this.removeKeyActions();
+
+        for (const direction in this.activeKeys) {
+            const newKey = customDirections[direction];
+
+            if (newKey && !this.activeKeys[direction].includes(newKey)) {

```

Comment:
I think that we can also check if the new key is string.
```suggestion
            if (typeof newKey === 'string' && !this.activeKeys[direction].includes(newKey)) {
```

Here if it is undefined it will fail.

Reply:
Done